swiftsimio/reader.py

```diff
@@ -624,10 +626,12 @@ class SWIFTDataset(object):
     SWIFTDataset.metadata.
     """
 
-    def __init__(self, filename, mask=None):
+    def __init__(self, filename, mask=None, generate_unit_func=metadata.unit_fields.generate_units,
+                 generate_cosmology_func=metadata.cosmology.cosmology_fields.generate_cosmology):
```

Comment:
Could you add type annotations here for callables? You can do this by writing `generate_unit_func:Callable=metadata.unit_fields.generate_units`